src/main/scala/io/iohk/ethereum/ledger/BloomFilter.scala

```diff
@@ -0,0 +1,55 @@
+package io.iohk.ethereum.ledger
+
+import akka.util.ByteString
+import io.iohk.ethereum.domain.TxLogEntry
+import io.iohk.ethereum.crypto._
+import io.iohk.ethereum.utils.ByteUtils.or
+
+object BloomFilter {
+
+  val BloomFilterByteSize: Int = 256
+  private val BloomFilterBitSize: Int = BloomFilterByteSize * 8
+  private val EmptyBloomFilter: Array[Byte] = Array.fill(BloomFilterByteSize)(0.toByte)
+  private val IntIndexesToAccess: Set[Int] = Set(0, 2, 4)
+
+  /**
+    * Given the logs of a receipt creates the bloom filter associated with them
+    * as stated in section 4.4.1 of the YP
+    *
+    * @param logs from the receipt whose bloom filter will be created
+    * @return bloom filter associated with the logs
+    */
+  def create(logs: Set[TxLogEntry]): ByteString = {
+    val bloomFilters = logs.map(createBloomFilterForLogEntry)
+    if(bloomFilters.isEmpty)
+      ByteString(EmptyBloomFilter)
+    else
+      ByteString(or(bloomFilters.toSeq: _*))
+  }
+
+  //Bloom filter function that reduces a log to a single 256-byte hash based on equation 24 of http://paper.gavwood.com/
+  private def createBloomFilterForLogEntry(logEntry: TxLogEntry): Array[Byte] = {
+    val dataForBloomFilter = logEntry.loggerAddress.bytes +: logEntry.logTopics
+    val bloomFilters = dataForBloomFilter.map(bytes => bloomFilter(bytes.toArray))
+
+    or(bloomFilters: _*)
+  }
+
+  //Bloom filter that sets 3 bits out of 2048 based on equations 25-28 of http://paper.gavwood.com/
+  private def bloomFilter(bytes: Array[Byte]): Array[Byte] = {
+    val hashedBytes = kec256(bytes)
+    val bitsToSet = IntIndexesToAccess.map{ i =>
+      ((hashedBytes(i + 1) & 0xFF) + ((hashedBytes(i) & 0xFF) << 8)) % BloomFilterBitSize
```

Comment:
You are right that in this line we get 11 bits from the 3 first pairs of bytes from kec256, 5 of the bits from the 16 bits are discarded while doing `% BloomFilterBitSize` (I tried to stick with how the formula is defined in the YP). I will add a comment and do some refactoring to make that clearer.